apps/squareone/next.config.js

```diff
@@ -45,11 +45,12 @@ const readServerYamlConfig = () => {
 module.exports = (phase, { defaultConfig }) => {
   const publicYamlConfig = readPublicYamlConfig();
   const serverYamlConfig = readServerYamlConfig();
+  const sentryDsn = process.env.SENTRY_DSN || null;
```

Comment:
A couple of alternatives to having this come from an env var:
* Have it come from another config file generated from a the new Kubernetes secret
* Have it plaintext in the Phalanx config (it's gonna be publicly visible on the Squareone frontend anyway)

Having it come from an env var allows us to keep it secret and seems simpler (though less consistent) than generating another config file.

Reply:
This seems like a reasonable approach. We had to use a YAML config file because many of the other configurations were harder to manage as env vars but this makes sense for secrets, which we don't have any existing examples of yet.

Do you have any concerns about the DSN being in the front end given that we manage it as a secret for the backend?

Reply:
There's no way around it for client-side code, it's expected to be shipped to the client. According to Sentry, there is not a lot of harm that can be done with a DSN:
https://docs.sentry.io/concepts/key-terms/dsn-explainer/

Keeping it as a secret slightly reduces its exposure by preventing it from being exposed in a GitHub repo somewhere (the Phalanx repo), but it's still available to anyone that visits any env's squareone site.  Given that, is it even worth making it a Phalanx secret?

Reply:
I guess the nice thing about it being a "secret" is that we get a single source of truth with our 1Password infrastructure. But on the other hand it's an extra secret to configure manually on USDF. So probably fine to keep it as it.